src/services/organizationService.ts

```diff
@@ -0,0 +1,209 @@
+import * as AR from "fp-ts/lib/Array";
+import { flow, pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { Organizations } from "../../generated/definitions/Organizations";
+import { OrganizationWithReferents } from "../../generated/definitions/OrganizationWithReferents";
+import {
+  Organization as OrganizationModel,
+  Referent,
+} from "../models/dbModels";
+import { Op, QueryTypes } from "sequelize";
+import { UpdateOrganizationPrimaryKey } from "../utils/postgres_queries";
+import { FiscalCode } from "@pagopa/ts-commons/lib/strings";
+import { Errors } from "io-ts";
+
+const filterByNameOrFiscalCode = (searchQuery?: string) =>
+  pipe(
+    O.fromNullable(searchQuery),
+    O.map((searchQuery) => ({
+      where: {
+        [Op.or]: [
+          { fiscal_code: { [Op.iLike]: `%${searchQuery}%` } },
+          { name: { [Op.iLike]: `%${searchQuery}%` } },
+        ],
+      },
+    })),
+    O.getOrElse(() => ({}))
+  );
+
+const paging = (page?: number, pageSize?: number) =>
+  pipe(
+    O.Do,
+    O.bind("page", () => O.fromNullable(page)),
+    O.bind("pageSize", () => O.fromNullable(pageSize)),
+    O.map(({ page, pageSize }) => ({
+      offset: page * pageSize,
+      limit: pageSize,
+    })),
+    O.getOrElse(() => ({}))
+  );
+
+export const getOrganizations = (
+  page?: number,
+  pageSize?: number,
+  searchQuery?: string
+): TE.TaskEither<Error, Organizations> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        OrganizationModel.findAll({
+          include: [OrganizationModel.associations.referents],
+          ...filterByNameOrFiscalCode(searchQuery),
+          ...paging(page, pageSize),
+        }),
+      E.toError
+    ),
+    TE.map(
+      flow(
+        AR.map((m) =>
+          OrganizationWithReferents.decode({
+            keyOrganizationFiscalCode: m.fiscalCode,
+            organizationFiscalCode: m.fiscalCode,
+            organizationName: m.name,
+            pec: m.pec,
+            insertedAt: m.insertedAt,
+            referents: m.referents.map((r) => r.fiscalCode),
+          })
+        ),
+        AR.filter(E.isRight),
+        AR.map((e) => e.right)
+      )
+    )
+  );
+
+export const upsertOrganization = (
+  organizationWithReferents: OrganizationWithReferents
+): TE.TaskEither<Error, OrganizationWithReferents> =>
+  pipe(
+    TE.of(
+      organizationWithReferents.keyOrganizationFiscalCode !==
+        organizationWithReferents.organizationFiscalCode
+    ),
+    TE.chain((shouldUpdatePrimaryKey) =>
+      TE.tryCatch(async () => {
```

Comment:
ditto for async

Reply:
in https://github.com/pagopa/ade-aa-ms-mock/pull/16/commits/9bd4e8454fe620851cd808de302de858df4fb62c